Data/ByteString/Streaming/Char8.hs

```diff
@@ -710,20 +711,215 @@ putStrLn :: MonadIO m => ByteString m r -> m r
 putStrLn bs = hPut IO.stdout (snoc bs '\n')
 {-# INLINE putStrLn #-}
 
--- | This will read positive or negative Ints that require 18 or fewer characters.
-readInt :: Monad m => ByteString m r -> m (Compose (Of (Maybe Int)) (ByteString m) r)
-readInt = go . toStrict . splitAt 18 where
-  go m = do
-    (bs :> rest) <- m
-    case Char8.readInt bs of
-      Nothing -> return (Compose (Nothing :> (chunk bs >> rest)))
-      Just (n,more) -> if B.null more
-        then do
-          e <- uncons rest
-          return $ case e of
-            Left r -> Compose (Just n :> return r)
-            Right (c,rest') -> if isDigit c
-               then Compose (Nothing :> (chunk bs >> cons' c rest'))
-               else Compose (Just n :> (chunk more >> cons' c rest'))
-        else return (Compose (Just n :> (chunk more >> rest)))
+-- | Bounds for Word# multiplication by 10 without overflow, and
+-- absolute values of Int bounds.
+intmaxWord, intminWord, intmaxQuot10, intmaxRem10, intminQuot10, intminRem10 :: Word
+intmaxWord = fromIntegral (maxBound :: Int)
+intminWord = fromIntegral (negate (minBound :: Int))
+(intmaxQuot10, intmaxRem10) = intmaxWord `quotRem` 10
+(intminQuot10, intminRem10) = intminWord `quotRem` 10
+
+-- Predicate to test whether a 'Word8' value is either ASCII whitespace,
+-- or a unicode NBSP (U+00A0).  Optimised for ASCII text, with spaces
+-- as the most frequent whitespace characters.
+w8IsSpace :: Word8 -> Bool
+w8IsSpace = \ !w8 ->
```

Comment:
A static bitmask literal `Addr#` will be probably faster. Cf. https://github.com/Bodigrim/unicode-transforms/blob/master/Data/Unicode/Internal/Bits.hs

Reply:
Yeah, I was looking for that, but found it tricky to do right, thanks for the link, I'll compare...

Reply:
I tested the bitmask code, it is nice, but actually slightly slower.  With just 3 cases to test (SP, NBSP and [TAB..CR]), the carefully tuned `w8IsSpace` is faster than the bitmask lookup, because just computing the word index and offset and then doing the lookup is somewhat slower.  The runtime for reading 100M space separated ints goes up from 3.8s to 4.0s with the bitmask code...  If the set of characters to include were more complex this would definitely be the way to go, but not in this case...

Reply:
Thanks, interesting. (For some reason I cannot resolve it myself)